knowledge_repo/app/models.py

```diff
@@ -26,6 +27,39 @@
 db_session = LocalProxy(lambda: current_app.db.session)
 
 
+class IndexMetadata(db.Model):
+    __tablename__ = 'index_metadata'
+
+    id = db.Column(db.Integer, nullable=False, primary_key=True)
+    type = db.Column(db.String(255), nullable=False)
```

Comment:
Yes; tuples of `type` and `name` should be unique. I'll encode this into the table as you suggest. We need type and name for things like:
type='repository_revision', name='<repository uri>', value='<revision>'

It would be possible to cram these into one field, but I felt like that was a bit clumsy. Agreed?